forest/cli/src/cli/snapshot_cmd.rs

```diff
@@ -249,79 +260,22 @@ fn remove(config: &Config, filename: &PathBuf, snapshot_dir: &Option<PathBuf>, f
 }
 
 fn prune(config: &Config, snapshot_dir: &Option<PathBuf>, force: bool) {
-    {
-        let snapshot_dir = snapshot_dir
-            .clone()
-            .unwrap_or_else(|| default_snapshot_dir(config));
-        println!("Snapshot dir: {}", snapshot_dir.display());
-        let mut snapshots_with_valid_name = vec![];
-        let mut snapshot_to_keep = None;
-        let mut latest_date = Date::MIN;
-        let mut latest_height = 0;
-        let pattern = Regex::new(
-            r"^forest_snapshot_([^_]+?)_(?P<date>\d{4}-\d{2}-\d{2})_height_(?P<height>\d+).car$",
-        )
-        .unwrap();
-        if let Ok(dir) = fs::read_dir(snapshot_dir) {
-            for path in dir
-                .flatten()
-                .map(|entry| entry.path())
-                .filter(|p| p.is_file())
-            {
-                if let Some(Some(filename)) = path.file_name().map(|n| n.to_str()) {
-                    if let Some(captures) = pattern.captures(filename) {
-                        let date = captures.name("date").unwrap();
-                        if let Ok(date) = time::Date::parse(date.as_str(), &Iso8601::DEFAULT) {
-                            let height = captures
-                                .name("height")
-                                .unwrap()
-                                .as_str()
-                                .parse::<i64>()
-                                .unwrap();
-                            if date > latest_date {
-                                latest_date = date;
-                                latest_height = height;
-                                snapshot_to_keep = Some(path.clone());
-                            } else if date == latest_date && height > latest_height {
-                                latest_height = height;
-                                snapshot_to_keep = Some(path.clone());
-                            }
-
-                            snapshots_with_valid_name.push(path);
-                        }
-                    } else if path.extension().unwrap_or_default() == "tmp" {
-                        snapshots_with_valid_name.push(path);
-                    }
-                }
-            }
-        }
-
-        if snapshots_with_valid_name.len() < 2 {
-            println!("No files to delete");
-            return;
-        }
-
-        let mut snapshots_to_delete = snapshots_with_valid_name;
-        let mut index_to_keep = 0;
-        println!("Files to delete:");
-        if let Some(snapshot_to_keep) = snapshot_to_keep {
-            for (i, path) in snapshots_to_delete.iter().enumerate() {
-                if &snapshot_to_keep != path {
-                    println!("{}", path.as_path().display());
-                } else {
-                    index_to_keep = i;
-                }
-            }
-        }
-        snapshots_to_delete.remove(index_to_keep);
+    let mut store = SnapshotStore::get_snapshots(config, snapshot_dir.clone()).unwrap();
```

Comment:
This `unwrap` is problematic. Fortunately, there's no reason why `get_snapshots` should return a `Result`.

Reply:
fixed